### DISCUSSION TITLE
Missing elevation

### DISCUSSION THREAD

**Discussion by Thomas-Lerman** [pelicanmapping/osgearth#2450] [Support Q&A] [answered]
Is there a way to tell when, for example, querying the elevation for a location from osgEarth whether a elevation file actually exists for that location? If not, other options? Hmmm, maybe I can grab the extents of each elevation file if they are available after I load them?

**Comment by gwaldron** [comment]
`ElevationPool::getSample` returns an `ElevationSample`, which in turn has a `hasData()` method that will return `true` if data exists at the sample location.

**Reply by Thomas-Lerman** [reply]
Thank you. Unfortunately, ElevationPool & getSample() seems to be in a newer version than we have. I did find hasData() used in ElevationLayer which is defined in TileSource. I have not dug deeper to see if I can use yet. Any other suggestions such as getting the extents of the files (although I could see this might not be perfect either) or something else?

**Comment by Thomas-Lerman** [comment, marked as answer]
I may have found an answer. It seems that if no elevation files exists, ElevationQuery::getElevation() returns the out_actualResolution as 0.0. It seems when at least one elevation file exists, but the point being queries is outside of the elevation file, out_actualResolution is returned as 5.8064517974853516
